src/planner/planner.c

```diff
@@ -1662,13 +1663,35 @@ cagg_reorder_groupby_clause(RangeTblEntry *subq_rte, Index rtno, List *outer_sor
 	}
 }
 
+/* Register our join pushdown hook. Becuase for PostgreSQL
+ * the tables we are operating on are local tables. So,
+ * the FDW hooks are not called. Register our join path
+ * generation as a generic planer hook.
+ */
+
+static void
+get_foreign_join_paths_hook(PlannerInfo *root, RelOptInfo *joinrel, RelOptInfo *outerrel,
```

Comment:
The name might be a little misleading, this is the join pathlist hook, not the `GetForeignJoinPaths` fdw method.